updated transfer does not raise ChannelOutdated anymore

hackaugusto opened the pull request.

LefterisJP commented:
@hackaugusto change is simple enough and looks okay. But why the change? What was wrong with it before? Is it documented anywhere?

hackaugusto commented:
@LefterisJP the exception is not raised anymore, so the `try-except` does not make sense. I don't believe this needs to be documented.

The raise was removed in another PR (here https://github.com/raiden-network/raiden/pull/3897 . Repeating what is on 3897 ) among the problems it fixed was:

- Made the code simpler
- Removed unnecessary extra RPC calls
- Fixed race conditions on pruned blocks
- Improved the exception messages. (which finally led to the reason why participant selection was flaky https://github.com/raiden-network/raiden/pull/3898 )

These changes should be explained in the relevant PR and in the commit messages.

LefterisJP commented:
I see. By documentation I meant a simple comment either in this commit message or in the PR. Essentially what you wrote here:

> the exception is not raised anymore, so the try-except does not make sense. I don't believe this needs to be documented.
